hudi-common/src/main/java/org/apache/hudi/common/table/HoodieTableMetaClient.java

```diff
@@ -232,6 +232,41 @@ public void buildIndexDefinition(HoodieIndexDefinition indexDefinition) {
     }
   }
 
+  /**
+   * Builds expression index definition and writes to index definition file.
+   */
+  public boolean buildColSatsIndexDefinition(HoodieIndexDefinition indexDefinition) {
+    String indexName = indexDefinition.getIndexName();
+    String indexMetaPath = getIndexDefinitionPath();
+    boolean updateIndexDefn = false;
+    if (indexMetadataOpt.isPresent()) {
+      if (indexMetadataOpt.get().getIndexDefinitions().containsKey(indexName)) {
```

Comment:
I think we can merge this method with the method above - `buildIndexDefinition`. At this point, I don't see a need to separate the two. Any special handling for columns stats can be done within the method. If at all a new method is necessary, I would still recommend to keep the file i/o related stuff in a common method, and just separate out the in-memory manipulation of index definition in separate methods. Main concern is reuse. Tomorrow, if we make any change in the protocol to update/delete the index definition file then we don't have to change at two places.

Reply:
sure. will take a look